src/lib/web-component/filter-timestamp/filter-timestamp-element.js

```diff
@@ -28,18 +31,15 @@ export class FilterTimestamp extends HTMLElement {
       return;
     }
 
-    // this.timestamp is null when timestamp filter is not applied.
-    if (!this.timeStamp) {
-      this.timeStamp = {};
-    }
-
```

Comment:
@rpl here is the part, I was talking about on the other comment. I can avoid this if I use a single type i.e. `object` only.

Reply:
@atiqueahmedziad it doesn't seem a big deal, you could also do something like this:

```js
const timeStamp = this.timeStamp || {}

....
if (info.menuItemId === 'startTime') {
  timeStamp.start = ...
  ...
} else if (info.menuItemId === 'stopTime') {
  timeStamp.stop = ...
  ...
}

...
this.timeStamp = timeStamp;
this.dispatchFilterChange();
```

and avoid the check `Object.keys(this.filter.timeStamp).length === 0`, which I personally don't like that much.

does this sounds a reasonable alternative for the issue you had?

Reply:
yes, this sounds good. But I had to use `Object.keys(this.filter.timeStamp).length === 0` while assigning empty object to null [here](https://github.com/mozilla/extension-activity-monitor/pull/40/commits/19f9545d2416104c5d450cbdb632b20328d06c55#diff-0d42cddec555ff90f252f9e7b5f94496R84).